src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -223,17 +223,17 @@ public void onGenerateMap() {
   private void handleGenerationException(Throwable e) {
     Throwable cause = e.getCause();
     if (cause instanceof InvalidParameterException) {
-      notificationService.addImmediateErrorNotification(e, "mapGenerator.invalidName");
       log.warn("Map generation failed", e);
+      notificationService.addImmediateErrorNotification(e, "mapGenerator.invalidName");
     } else if (cause instanceof UnsupportedVersionException) {
-      notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooNewVersion");
       log.warn("Map generation failed", e);
+      notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooNewVersion");
     } else if (cause instanceof OutdatedVersionException) {
-      notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooOldVersion");
       log.warn("Map generation failed", e);
+      notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooOldVersion");
     } else {
-      notificationService.addImmediateErrorNotification(e, "mapGenerator.generationFailed");
       log.warn("Map generation failed", e);
+      notificationService.addImmediateErrorNotification(e, "mapGenerator.generationFailed");
     }
   }
```

Comment:
All of these warning should either be moved outside the if block or customized to say more about the type of error.